base_local_planner/src/obstacle_cost_function.cpp

```diff
@@ -160,7 +161,11 @@ double ObstacleCostFunction::footprintCost (
     return -7.0;
   }
 
-  double occ_cost = std::max(std::max(0.0, footprint_cost), double(costmap->getCost(cell_x, cell_y)));
+  double occ_cost = double(costmap->getCost(cell_x, cell_y));
+  if (occdist_use_footprint_)
+  {
+    occ_cost = std::max(std::max(0.0, footprint_cost), occ_cost);
+  }
 
   return occ_cost;
```

Comment:
What if `occ_cost` is lethal? does it make sense to return `< 0` value then?

Reply:
That is already done a few lines above:

https://github.com/rapyuta-robotics/ros-navigation/blob/1705c5eb553076abc402262bccc4fb198bdf90b2/base_local_planner/src/obstacle_cost_function.cpp#L152-L156

Reply:
Yeah, but isn't that applicable to only `footprint_cost`? What about at just at `(x, y)` i.e. `costmap->getCost(cell_x, cell_y)`?

Reply:
Hm yes, but I don't think that will ever happen in reality -- it would mean there is an obstacle completely enveloped by the robot.

But I didn't change the logic in this PR, I simply moved part of the existing code inside an if statement. If we want to address what you suggest I think it would be best to do in a different PR.